torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -354,21 +354,21 @@ class DeviceContextArena {
   }
 
   torch::lazy::BackendDataPtr GetRngSeedData(
-      const torch::lazy::BackendDevice& device, bool reset) {
+      const torch::lazy::BackendDevice& device) {
     static const at::ScalarType kSeedType = at::ScalarType::Long;
+
+    // make these consts match with GetRngSeed method
+    static const uint64_t kSeedMul = 214013;
+    static const uint64_t kSeedAdd = 2531011;
     DeviceContext* devctx = GetDeviceContext(device);
     std::lock_guard<std::mutex> lock(devctx->lock);
-    if (reset) {
-      devctx->seed = 1012031 + devctx->seed * 7012063;
-      devctx->running_seed = devctx->seed;
-      at::Tensor tensor = at::scalar_tensor(MakeIntScalar(devctx->seed),
-                                            at::TensorOptions(kSeedType));
-      torch::lazy::BackendDataPtr device_data = TensorToXlaData(tensor, device);
-      devctx->seed_ir_value = torch::lazy::MakeNode<DeviceData>(device_data);
-    } else if (!devctx->seed_ir_value) {
-      devctx->seed_ir_value =
-          IrValueFromScalar(MakeIntScalar(devctx->seed), kSeedType, device);
-    }
+
+    devctx->running_seed = kSeedAdd + kSeedMul * devctx->running_seed;
```

Comment:
we force the running seed and real seed divergeg here, which sounds like a bad thing..

Reply:
This is to be equivalent to what regular torchxla does. In regular torchxla, when a seed is needed, it calls [XLAGraphExecutor::GetRngSeed](https://github.com/pytorch/xla/blob/master/torch_xla/csrc/xla_graph_executor.cpp#L319) to grab the seed IR node. We need do the same thing for the dynamo/torchxla bridge. The only difference is we need a DeviceData node for the bridge.

Increment the running seed so that the next call will get a different seed. This also what `XLAGraphExecutor::GetRngSeed` does.

Let me know if there is more clarification needed

Reply:
In the `GetRngSeed`, we incremented both `devctx->running_seed` and `devctx->seed_ir_value`. However in this function we only increment `devctx->running_seed`.

Imagine a user call this `GetRngSeedData` function multiple times, it will always get the same result because `devctx->seed` is not being updated. 

If the purpose of this function is to get the current base seed without incremented it(since we expect `mark_step` to do so) , we should not touch the `running_seed` and rename this function to `GetBaseRngSeedData`. We also should overwrite `devctx->seed_ir_value` but just return the base seed as an data ptr

Reply:
I got what you mean. I think the issue is at line 368, we should use devctx->running_seed rather than devctx->seed. Will send an update and some test case soon.